src/Model/Problemlayer/Timeevolver.cpp

```diff
@@ -65,12 +67,14 @@ namespace Model {
       retries(timeevolver_data["retries"]),
       use_simplified_newton(timeevolver_data["use_simplified_newton"]) {}
 
-  void Timeevolver::simulate(
+  std::chrono::duration<double> Timeevolver::simulate(
```

Comment:
Is there a benefit of making Timeevolver do the measurements?
If not I would rather have the time keeping separate from the actual computations and just take a new
`systemclock::now()` after `simulate(...)` completed inside commands.txt.

Reply:
no - that is just legacy - I did not move it from there when implementing the other timers - will fix

Reply:
actually I remembered: when the time is measured inside the Timeevolver then the deallocation which happens at the end of the function is happening after the time was measured which allows for the "teardown" timer. But since that is just milliseconds it is probably not worth it.

Reply:
Yes, not much is destroyed at the end of simulate. Its a couple of doubles and two Eigen::VectorXd.